pkg/controller/garbagecollector/references/references.go

```diff
@@ -0,0 +1,153 @@
+	}
+
+	kind = split[0]
+	name = split[1]
+	return
+}
+
+// InjectAnnotations injects annotations into the annotation maps based on the referenced ConfigMaps/Secrets appearing
+// in the pod template spec's `.volumes[]` or `.containers[].envFrom[]` lists. Additional reference annotations can be
+// specified via the variadic parameter.
+func InjectAnnotations(obj runtime.Object, additional ...string) {
+	switch o := obj.(type) {
+	case *corev1.Pod:
+		referenceAnnotations := computeAnnotations(o.Spec, additional...)
+		o.Annotations = utils.MergeStringMaps(o.Annotations, referenceAnnotations)
+
+	case *appsv1.ReplicaSet:
+		referenceAnnotations := computeAnnotations(o.Spec.Template.Spec, additional...)
+		o.Annotations = utils.MergeStringMaps(o.Annotations, referenceAnnotations)
+		o.Spec.Template.Annotations = utils.MergeStringMaps(o.Spec.Template.Annotations, referenceAnnotations)
+
+	case *appsv1.Deployment:
+		referenceAnnotations := computeAnnotations(o.Spec.Template.Spec, additional...)
+		o.Annotations = utils.MergeStringMaps(o.Annotations, referenceAnnotations)
+		o.Spec.Template.Annotations = utils.MergeStringMaps(o.Spec.Template.Annotations, referenceAnnotations)
+
+	case *appsv1.StatefulSet:
+		referenceAnnotations := computeAnnotations(o.Spec.Template.Spec, additional...)
+		o.Annotations = utils.MergeStringMaps(o.Annotations, referenceAnnotations)
+		o.Spec.Template.Annotations = utils.MergeStringMaps(o.Spec.Template.Annotations, referenceAnnotations)
+
+	case *appsv1.DaemonSet:
+		referenceAnnotations := computeAnnotations(o.Spec.Template.Spec, additional...)
+		o.Annotations = utils.MergeStringMaps(o.Annotations, referenceAnnotations)
+		o.Spec.Template.Annotations = utils.MergeStringMaps(o.Spec.Template.Annotations, referenceAnnotations)
+
+	case *batchv1.Job:
+		referenceAnnotations := computeAnnotations(o.Spec.Template.Spec, additional...)
+		o.Annotations = utils.MergeStringMaps(o.Annotations, referenceAnnotations)
+		o.Spec.Template.Annotations = utils.MergeStringMaps(o.Spec.Template.Annotations, referenceAnnotations)
+
+	case *batchv1.CronJob:
+		referenceAnnotations := computeAnnotations(o.Spec.JobTemplate.Spec.Template.Spec, additional...)
+		o.Annotations = utils.MergeStringMaps(o.Annotations, referenceAnnotations)
+		o.Spec.JobTemplate.Annotations = utils.MergeStringMaps(o.Spec.JobTemplate.Annotations, referenceAnnotations)
+		o.Spec.JobTemplate.Spec.Template.Annotations = utils.MergeStringMaps(o.Spec.JobTemplate.Spec.Template.Annotations, referenceAnnotations)
+
+	case *batchv1beta1.CronJob:
+		referenceAnnotations := computeAnnotations(o.Spec.JobTemplate.Spec.Template.Spec, additional...)
+		o.Annotations = utils.MergeStringMaps(o.Annotations, referenceAnnotations)
+		o.Spec.JobTemplate.Annotations = utils.MergeStringMaps(o.Spec.JobTemplate.Annotations, referenceAnnotations)
+		o.Spec.JobTemplate.Spec.Template.Annotations = utils.MergeStringMaps(o.Spec.JobTemplate.Spec.Template.Annotations, referenceAnnotations)
+	}
```

Comment:
return an error here if the given object GVK isn't handled. It's better to fail explicitly than do nothing silently.

Reply:
Not sure, I would like to avoid error handling for the gazillion places this function will be used. The doc string already states what the function does, so passing an object which does not have any pod template spec does not make sense anyways.